src/ansi-c/expr2c.cpp

```diff
@@ -378,7 +383,10 @@ std::string expr2ct::convert_rec(
 
     dest+=" }";
 
-    dest+=d;
+    if(src.find(ID_typedef).is_not_nil())
+      dest+=src.get_string(ID_typedef);
+    else
+      dest+=d;
```

Comment:
So this seems unrelated as I get this problem on the main branch, I have created an issue #355 and a PR that includes a regression test checking for it: #356